qiita_db/handlers/processing_job.py

```diff
@@ -0,0 +1,157 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from datetime import datetime
+from json import loads
+
+from tornado.web import RequestHandler
+
+import qiita_db as qdb
+
+
+def _get_job(job_id):
+    """Returns the job with the given id if it exists
+
+    Parameters
+    ----------
+    job_id : str
+        The job id to check
+
+    Returns
+    -------
+    qiita_db.processing_job.ProcessingJob, bool, string
+        The requested job or None
+        Whether if we could get the job or not
+        Error message in case we couldn't get the job
+    """
+    if not qdb.processing_job.ProcessingJob.exists(job_id):
+        return None, False, 'Job does not exist'
+
+    try:
+        job = qdb.processing_job.ProcessingJob(job_id)
+    except qdb.exceptions.QiitaDBError as e:
+        return None, False, 'Error instantiating the job: %s' % str(e)
+
+    return job, True, ''
+
+
+class JobHandler(RequestHandler):
+    def get(self, job_id):
+        """Retrieves the work that need to be done for the job
+
+        Parameters
+        ----------
+        job_id : str
+            The job that the information needs to be returned
+        """
+        with qdb.sql_connection.TRN:
+            job, success, error_msg = _get_job(job_id)
+            cmd = None
+            params = None
+            status = None
+            if success:
+                cmd = job.command.name
+                params = job.parameters.values
+                status = job.status
+
+        response = {'success': success, 'error': error_msg,
+                    'command': cmd, 'parameters': params,
+                    'status': status}
+        self.write(response)
+
+
+class HeartbeatHandler(RequestHandler):
+    def post(self, job_id):
+        """Perform a heartbeat against a job
+
+        Parameters
+        ----------
+        job_id : str
+            The job to perform the heartbeat
+        """
+        with qdb.sql_connection.TRN:
+            job, success, error_msg = _get_job(job_id)
+            if success:
+                job_status = job.status
+                if job_status == 'queued':
+                    job.status = 'running'
+                    job.heartbeat = datetime.now()
+                elif job_status == 'running':
+                    job.heartbeat = datetime.now()
+                else:
+                    success = False
+                    error_msg = 'Job already finished. Status: %s' % job_status
+
+        response = {'success': success, 'error': error_msg}
+        self.write(response)
+
+
+class StepHandler(RequestHandler):
+    def post(self, job_id):
+        """Changes the step of the given job
```

Comment:
I didn't think about this before, but what happens when there are multiple jobs running at the moment i.e. you've started a meta analysis and the analysis creation began, now you have three parallel jobs summarize taxa, beta diversity and alpha, would this handler report that only one of them is the "current step" or would it handle this correctly? It seems like it would only report one, right?